project/webapp/controller/ProductsList.controller.js

```diff
@@ -1,15 +1,203 @@
+
+          oTableBinding.filter(aFilters);
+        },
+
+        onClearFilters() {
+          const oTableBinding = this.byId("productList").getBinding("items");
+
+          this.byId("releaseDate").setValue(null);
+          this.byId("supplier").setTokens([]);
+          this.byId("searchName").setValue(null);
+          this.byId("categorySelect").setSelectedKeys([]);
+
+          oTableBinding.filter(null);
+        },
+
+        _getSearchNameFilter() {
+          const sSearchQuery = this.byId("searchName").getProperty("value");
+
+          return sSearchQuery.length
+            ? new Filter("name", FilterOperator.Contains, sSearchQuery)
+            : null;
+        },
+
+        _getCategoriesFilter() {
+          const aSelectedCategories =
+            this.byId("categorySelect").getProperty("selectedKeys");
+          let aFilters = [];
+
+          if (!!aSelectedCategories.length) {
+            aFilters = aSelectedCategories.map((sSelectedKey) => {
+              return new Filter({
+                path: "categories",
+                operator: FilterOperator.EQ,
+                value1: sSelectedKey,
+                test: (aCategories) => {
+                  return aCategories.some(({ id }) => id === sSelectedKey);
+                }
+              });
+            });
+          }
+
+          return aFilters.length ? aFilters : null;
+        },
+
+        _getDateFilter() {
+          const sDate = this.byId("releaseDate");
+          const sDateStart = sDate.getDateValue();
+          const sDateEnd = sDate.getSecondDateValue();
+
+          if (!sDateStart || !sDateEnd) {
+            return null;
+          }
+
+          return new Filter({
+            path: "releaseDate",
+            operator: FilterOperator.BT,
+            value1: sDateStart.toISOString(),
+            value2: sDateEnd.toISOString()
+          });
+        },
+
+        _getCurrentTokens(oEvent) {
+          let aSuppliersTokens = this.byId("supplier").getTokens();
+          const oEventParameters = oEvent.getParameters();
+          const sTokenType = oEventParameters.type;
+
+          if (sTokenType === this.TOKEN_REMOVED_TYPE) {
+            const sRemovedTokenKey = oEventParameters.removedTokens[0].getProperty("key");
+
+            aSuppliersTokens = aSuppliersTokens.filter(
+              (item) => item.getProperty("key") !== sRemovedTokenKey
+            );
+          }
+
+          return aSuppliersTokens;
+        },
+
+        _getSupplierFilterWithoutToken() {
+          const sSupplierValue = this.byId("supplier").getValue();
+
+          return [
+            new Filter({
+              path: "suppliers",
+              operator: FilterOperator.EQ,
+              value1: sSupplierValue,
+              test: () => sSupplierValue === ""
+            })
+          ];
+        },
+
+        _getSupplierFilterWithTokens(aSuppliersTokens) {
+          const aFilters = aSuppliersTokens.map((oToken) => {
```

Comment:
you can do not create aFilters variable, just write return aSuppliersTokens.map((oToken) => ....